File: leetsync/3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.go
func minOperations(nums []int) int {
	n := len(nums)
	k := 3
	isFlipped := make([]bool, n)
	totalFlips := 0
	flipCountFromPastForCurrentI := 0
	for i := 0; i < n; i++ {
		if i >= k && isFlipped[i-k] {
			flipCountFromPastForCurrentI--
		}
		if flipCountFromPastForCurrentI%2 == nums[i] {
			if (i + k) > n {
				return -1
			}
			totalFlips++
			isFlipped[i] = true
			flipCountFromPastForCurrentI++
		}
	}
	return totalFlips
}

File: leetsync/3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.java
class Solution {
    public int minOperations(int[] nums) {
        int n = nums.length;
        int k = 3;
        boolean[] isFlipped = new boolean[n];
        int totalFlips = 0;
        int flipCountFromPastForCurrentI = 0;
        for (int i = 0; i < n; i++) {
            if (i >= k && isFlipped[i - k]) {
                flipCountFromPastForCurrentI--;
            }
            if (flipCountFromPastForCurrentI % 2 == nums[i]) {
                if (i + k > n) {
                    return -1;
                }
                totalFlips++;
                isFlipped[i] = true;
                flipCountFromPastForCurrentI++;
            }
        }
        return totalFlips;
    }
}